plugin/storage/grpc/factory.go

```diff
@@ -49,6 +48,24 @@ type Factory struct {
 	remoteConn     *grpc.ClientConn
 }
 
+type host struct {
+	extensions map[component.ID]component.Component
+}
```

Comment:
@chahatsagarmain so we actually don't need to define a host at all. if you take a look at [this method](https://github.com/jaegertracing/jaeger/pull/6058), you'll see that the host is available to us already. What you need to do is pass the host down from the method I linked above into [this](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L129) function and you can then pass it into `Initialize` and then into `ToClientConn`. Does that make sense?

Reply:
Passing as a params to Initialize gives declaration error for Factory but Creating a host in `NewFactoryWithConfig ` wouldnt that also be a no op host?

Reply:
You'll need to change the interface in https://github.com/jaegertracing/jaeger/blob/main/storage/factory.go#L30 for this to work. You won't be be creating a host in `NewFactoryWithConfig`, you'll be passing it into that method (which will then pass it to initialize. )

Reply:
@yurishkuro adding `component.Host` to Initialize would break the `storage.Factory` interface. Is this something we want to do?

Reply:
@mahadzaryab1 can the otel collector initialize a `component.Host` in `collectorParams` which can be used here ?

Reply:
I think you'll want to do something like this: 

1. Update factory struct to contain the host 
```go
type Factory struct {
	metricsFactory metrics.Factory
	logger         *zap.Logger
	tracerProvider trace.TracerProvider
	config         Config
	services       *ClientPluginServices
	remoteConn     *grpc.ClientConn
	host           component.Host
}
```
2. Update `NewFactoryWithConfig` so that it takes in the host and sets it in the factory
```go
func NewFactoryWithConfig(
	cfg Config,
	metricsFactory metrics.Factory,
	logger *zap.Logger,
	host component.Host,
) (*Factory, error) {
	f := NewFactory()
	f.config = cfg
	f.host = host
	if err := f.Initialize(metricsFactory, logger); err != nil {
		return nil, err
	}
	return f, nil
}
```
3. Use `f.host` in `Initialize`

Reply:
+1 - don't change the interface, but ok to change constructor